### PR TITLE
Add support for importing markdown in storybook

### PR DESCRIPTION
**Pull request by dangreenisrael**
Issue:

## What I did
Added support for markdown (parse to html string) in react storybook at the webpack level

## How to test
Import a markdown file and assert that it has paresed as expected

Is this testable with jest or storyshots?
No

Does this need a new example in the kitchen sink apps?
Yes.

Does this need an update to the documentation?
Yes.

## Screenshot from public site
screen shot 2017-11-15 at 9 15 42 am

### CODE REVIEW

**Comment by ndelangen** [comment]
Would this work with your addon @tuchk4 ?

**Comment by tuchk4** [comment]
@ndelangen Seems no because for [readme-addon](https://github.com/tuchk4/storybook-readme) *\*.md* files should be loaded as raw.

```js
rules: [{
  test: /\.md$/,
  use: "raw-loader"
}]
```

---
I think I can update addon to work with html (md -> html transform via loaders) as suggested in this PR. It is good because there will no additional webpack setup for addon.

Why do you need md loader at storybook core?

**Comment by dangreenisrael** [comment]
@tuchk4 This all started with someone passing in a string of HTML into notes and getting a bunch of <br/> tags added (https://github.com/storybooks/storybook/issues/2005). I had originally wanted the ability to use JSX in notes, but that is a non starter because things need to be serialized to go in addon channels. Defining markdown in a javascript string is fraught with problems, so importing it seemed like the best option. The reason behind doing it in storybook core is that it would provide a simple, standardized way for people to use markdown in storybook. 

### Using markdown in storybook
1. Import a markdown file
2. Do something with the resulting HTML string

**Comment by dangreenisrael** [comment]
@tuchk4 I think its worth mentioning that your plugin looks AWESOME, and I'm going to looking into implementing at wave.

**Comment by tuchk4** [comment]
@dangreenisrael Today I made huge update and now `storybook-readme` addon is compatible with this PR. I removed `raw-loader` and added `html` and `markdown` loaders. 
If merge this PR to storybook core - users should not make additoinal webpack configuration for `sotrybook-readme`.  This is awesome! :)

**Comment by dangreenisrael** [comment]
@tuchk4 That is awesome! @ndelangen If there is support for this, I can expand this to work for all of the platforms (not just react). I wouldn't be able to do it for at least a few days though. I don't think that this should be merged in yet, I just opened this PR to show the concept.

**Comment by ndelangen** [comment]
@tuchk4 Maybe it's time to merge your addon into the monorepo?

**Review by ndelangen — APPROVED** [review]

**Review by danielduan — CHANGES_REQUESTED** [review]
Please add documentation where you think is appropriate.

I believe `withNotes` should document this along with some central location for the overall storybook package. Perhaps the docs on storybook.js.org too.

**Comment by tuchk4** [comment]
@ndelangen lets merge it this week

**Comment by dangreenisrael** [comment]
@danielduan I added to the withNotes docs. I'm not sure where the best universal place to add it is. Also, are we adding docs to the new-docs branch or to release/3.3?

**Comment by danielduan** [comment]
@dangreenisrael if this is getting merged to `release/3.3`, I believe that's where it should go as well.

File: addons/notes/README.md [hunk 1]
```diff
@@ -41,9 +41,23 @@ import { withNotes } from '@storybook/addon-notes';
...
 
+#### Using Markdown
+
+To use markdown in your notes simply import a markdown file and use that in your note.
+
+```js
+import { storiesOf } from '@storybook/react';
+import { withNotes } from '@storybook/addon-notes';
+import Component from './Component';
+import someMarkdownText from './someMarkdownText';
```
Comment (ndelangen) [inline addons/notes/README.md]: Needs a `.md` extension.

**Review by ndelangen — CHANGES_REQUESTED** [review]

**Review by ndelangen — APPROVED** [review]

**Comment by Hypnosphi** [comment]
Looks like `addon-info` has its own markdown support: https://github.com/storybooks/storybook/tree/master/addons/info#basic-usage

I'd prefer to have some consistency in how info and notes addons handle input

**Comment by dangreenisrael** [comment]
@Hypnosphi. The idea is more to allow users to import markdown files as HTML in general, and less to have markdown supported by notes.  This also allows @tuchk4 ‘s addon to work out of the box, without users having to worry about webpack. More generally, I think it’s expected behaviour with a tool like storybook to be able to import a markdown file.

**Review by danielduan — APPROVED** [review]

**Comment by danielduan** [comment]
I think the difference is that if you have a markdown file in your codebase already, you should be able to use notes and import it. But if you're trying to write some documentation about a component in markdown just for Storybook, you can include it in info as a string.

**Pull request merged**